### PR TITLE
Setup travis.yaml to check tests and format

### PR DESCRIPTION
**Pull request by rockneurotiko**
The good news:
- Travis will require to pass the tests
- Travis will require the format

The bad news:
Because the format are on elixir 1.6, we need to setup the travis.yml to use that version.

I changed the mix.exs vesion too, to ensure that there are no incompatibilities, but maybe you want to let the mix.exs version at 1.2 and pass the tests & format on 1.6.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 otp_release:
-  - 18.0
+  - 20.2
+cache:
+  directories:
+    - _build
+    - deps
+script:
+  - mix test || travis_terminate 1
+  - mix format --check-formatted || travis_treminate 1
```
Comment (anronin) [inline .travis.yml]: typo here travis_te**r**minate

Comment (rockneurotiko) [reply .travis.yml]: woops thanks!

**Comment by zhyu** [comment]
Oh, I like this pr!
Thank you!

I'm going to release version 0.4.3 which contains several patches, and merge this pr later.
So users can have those patches without upgrading to elixir 1.6.

**Comment by zhyu** [comment]
close #62

**Comment by zhyu** [comment]
close #60

**Pull request merged**